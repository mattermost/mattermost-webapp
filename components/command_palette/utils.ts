// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {Channel} from '@mattermost/types/channels';
import {Team} from '@mattermost/types/teams';

// See LICENSE.txt for license information.
import {CommandPaletteItem} from './command_palette_list_item/command_palette_list_item';
import {
    CmdPalettePictographIcon,
    CmdPalettePictographType,
} from './command_palette_list_item/command_palette_list_item_pictograph';
import {ChannelType, CommandPaletteEntities} from './types';

const channelTypeOrder = {
    [ChannelType.OPEN_CHANNEL]: 0,
    [ChannelType.PRIVATE_CHANNEL]: 1,
    [ChannelType.DM_CHANNEL]: 2,
    [ChannelType.GM_CHANNEL]: 3,
};

function filterName(name: string): string {
    return name.replace(/[.,'"\/#!$%\^&\*;:{}=\-_`~()]/g, ''); // eslint-disable-line no-useless-escape
}

export function channelToCommandPaletteItemTransformer(channels: Channel[], teams: Record<string, Team>): CommandPaletteItem[] {
    return channels.map((channel) => {
        const isArchived = channel.delete_at ? channel.delete_at !== 0 : false;

        let channelIcon = CmdPalettePictographIcon.GLOBE;
        if (isArchived) {
            channelIcon = CmdPalettePictographIcon.ARCHIVE;
        }
        return {
            description: channel.type === ChannelType.DM_CHANNEL ? channel.name : '',
            id: channel.id,
            isDeactivated: isArchived,
            teamName: teams[channel.team_id]?.name,
            pictograph: {
                type: CmdPalettePictographType.ICON,
                pictographItem: channelIcon,
            },
            subType: channel.type as ChannelType,
            title: channel.display_name,
            type: CommandPaletteEntities.Channel,
            isArchived,
        };
    });
}

function sortChannelsByTypeAndDisplayName(locale: string, cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem): number {
    if (channelTypeOrder[cmdPaletteItemA.subType] !== channelTypeOrder[cmdPaletteItemB.subType]) {
        if (channelTypeOrder[cmdPaletteItemA.subType] < channelTypeOrder[cmdPaletteItemB.subType]) {
            return -1;
        }

        return 1;
    }

    const aDisplayName = filterName(cmdPaletteItemA.title);
    const bDisplayName = filterName(cmdPaletteItemB.title);

    if (aDisplayName !== bDisplayName) {
        return aDisplayName.toLowerCase().localeCompare(bDisplayName.toLowerCase(), locale, {numeric: true});
    }

    return cmdPaletteItemA.title.toLowerCase().localeCompare(cmdPaletteItemA.title.toLowerCase(), locale, {numeric: true});
}

function sortChannelsByRecencyAndTypeAndDisplayName(cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem) {
    if (cmdPaletteItemA.lastViewedAt && cmdPaletteItemB.lastViewedAt) {
        return cmdPaletteItemB.lastViewedAt - cmdPaletteItemA.lastViewedAt;
    } else if (cmdPaletteItemA.lastViewedAt) {
        return -1;
    } else if (cmdPaletteItemB.lastViewedAt) {
        return 1;
    }

    // MM-12677 When this is migrated this needs to be fixed to pull the user's locale
    return sortChannelsByTypeAndDisplayName('en', cmdPaletteItemA, cmdPaletteItemB);
}

export function commandPaletteSorter(prefix: string, cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem) {
    if (cmdPaletteItemA.isArchived && !cmdPaletteItemB.isArchived) {
        return 1;
    } else if (!cmdPaletteItemA.isArchived && cmdPaletteItemB.isArchived) {
        return -1;
    }

    if (cmdPaletteItemA.isDeactivated && !cmdPaletteItemB.isDeactivated) {
        return 1;
    } else if (cmdPaletteItemB.isDeactivated && !cmdPaletteItemA.isDeactivated) {
        return -1;
    }

    let aDisplayName = cmdPaletteItemA.title.toLowerCase();
    let bDisplayName = cmdPaletteItemA.title.toLowerCase();

    if (cmdPaletteItemA.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemA.subType === ChannelType.DM_CHANNEL &&
        aDisplayName.startsWith('@')) {
        aDisplayName = aDisplayName.substring(1);
    }

    if (cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemB.subType === ChannelType.DM_CHANNEL && bDisplayName.startsWith('@')) {
        bDisplayName = bDisplayName.substring(1);
    }

    const aStartsWith = aDisplayName.startsWith(prefix) || cmdPaletteItemA.description.toLowerCase().startsWith(prefix);
    const bStartsWith = bDisplayName.startsWith(prefix) || cmdPaletteItemB.description.toLowerCase().startsWith(prefix);

    // Open channels user haven't interacted should be at the  bottom of the list
    if (cmdPaletteItemA.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemA.subType === ChannelType.OPEN_CHANNEL &&
        !cmdPaletteItemA.lastViewedAt &&
        ((cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
            cmdPaletteItemB.subType !== ChannelType.OPEN_CHANNEL) || cmdPaletteItemB.lastViewedAt)) {
        return 1;
    } else if ((cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemB.subType === ChannelType.OPEN_CHANNEL) || !cmdPaletteItemB.lastViewedAt) {
        return -1;
    }

    // Sort channels starting with the search term first
    if (aStartsWith && !bStartsWith) {
        return -1;
    } else if (!aStartsWith && bStartsWith) {
        return 1;
    }
    return sortChannelsByRecencyAndTypeAndDisplayName(cmdPaletteItemA, cmdPaletteItemB);
}
