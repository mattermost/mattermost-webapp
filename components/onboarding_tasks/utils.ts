// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const generateTelemetryTag = (category: string, name: string, suffix: string) => {
    return `${category}--${name}--${suffix}`;
};

export function getAnalyticsCategory(isAdmin: boolean) {
    return isAdmin ? 'admin_onboarding' : 'end_user_onboarding';
}
