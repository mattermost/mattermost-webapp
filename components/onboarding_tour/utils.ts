// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {FINISHED} from './constant';

export const KeyCodes: Record<string, [string, number]> = {
    ENTER: ['Enter', 13],
    COMPOSING: ['Composing', 229],
};

// This is extracted from utils file to remove dependency on utils file of webapp
export function isKeyPressed(event: KeyboardEvent, key: [string, number]): boolean {
    // There are two types of keyboards
    // 1. English with different layouts(Ex: Dvorak)
    // 2. Different language keyboards(Ex: Russian)

    if (event.keyCode === KeyCodes.COMPOSING[1]) {
        return false;
    }

    // checks for event.key for older browsers and also for the case of different English layout keyboards.
    if (typeof event.key !== 'undefined' && event.key !== 'Unidentified' && event.key !== 'Dead') {
        const isPressedByCode = event.key === key[0] || event.key === key[0].toUpperCase();
        if (isPressedByCode) {
            return true;
        }
    }

    // used for different language keyboards to detect the position of keys
    return event.keyCode === key[1];
}

export const getLastStep = (Steps: Record<string, number>) => {
    return Object.values(Steps).reduce((maxStep, candidateMaxStep) => {
        // ignore the "opt out" FINISHED step as the max step.
        if (candidateMaxStep > maxStep && candidateMaxStep !== FINISHED) {
            return candidateMaxStep;
        }
        return maxStep;
    }, Number.MIN_SAFE_INTEGER);
};
