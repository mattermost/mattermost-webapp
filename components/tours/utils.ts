// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {FINISHED} from './constant';

export type ActionType = 'next' | 'prev' | 'dismiss' | 'jump' | 'skipped'

export interface ChannelsTourTipManager {
    show: boolean;
    currentStep: number;
    tourSteps: Record<string, number>;
    handleOpen: (e: React.MouseEvent) => void;
    handleSkip: (e: React.MouseEvent) => void;
    handleDismiss: (e: React.MouseEvent) => void;
    handlePrevious: (e: React.MouseEvent) => void;
    handleNext: (e: React.MouseEvent) => void;
    handleJump: (e: React.MouseEvent, jumpStep: number) => void;
}

export const KeyCodes: Record<string, [string, number]> = {
    ENTER: ['Enter', 13],
    COMPOSING: ['Composing', 229],
};

// This is extracted from utils file to remove dependency on utils file of webapp
export function isKeyPressed(event: KeyboardEvent, key: [string, number]): boolean {
    // There are two types of keyboards
    // 1. English with different layouts(Ex: Dvorak)
    // 2. Different language keyboards(Ex: Russian)

    if (event.keyCode === KeyCodes.COMPOSING[1]) {
        return false;
    }

    // checks for event.key for older browsers and also for the case of different English layout keyboards.
    if (typeof event.key !== 'undefined' && event.key !== 'Unidentified' && event.key !== 'Dead') {
        const isPressedByCode = event.key === key[0] || event.key === key[0].toUpperCase();
        if (isPressedByCode) {
            return true;
        }
    }

    // used for different language keyboards to detect the position of keys
    return event.keyCode === key[1];
}

export const getLastStep = (Steps: Record<string, number>) => {
    return Object.values(Steps).reduce((maxStep, candidateMaxStep) => {
        // ignore the "opt out" FINISHED step as the max step.
        if (candidateMaxStep > maxStep && candidateMaxStep !== FINISHED) {
            return candidateMaxStep;
        }
        return maxStep;
    }, Number.MIN_SAFE_INTEGER);
};
