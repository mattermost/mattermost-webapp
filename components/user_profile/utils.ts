// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import ColorHash from 'color-hash';
import ColorContrastChecker from 'color-contrast-checker';

const cachedColors = new Map<string, string>();

export function generateColor(username: string, background: string): string {
    const cacheKey = `${username}-${background}`;
    const cachedColor = cachedColors.get(cacheKey);
    if (cachedColor) {
        return cachedColor;
    }

    let userColor = background;
    let userAndSalt = username;
    const checker = new ColorContrastChecker();
    const colorHash = new ColorHash();

    let tries = 3;
    while (!checker.isLevelCustom(userColor, background, 4.5) && tries > 0) {
        userColor = colorHash.hex(userAndSalt);
        userAndSalt += 'salt';
        tries--;
    }

    cachedColors.set(cacheKey, userColor);
    return userColor;
}
