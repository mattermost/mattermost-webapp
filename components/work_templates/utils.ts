// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {PlaybookTemplateType, WorkTemplate} from '@mattermost/types/work_templates';

export function getTemplateDefaultIllustration(template: WorkTemplate): string {
    const channels = template.content.filter((c) => c.channel).map((c) => c.channel!);
    if (channels.length) {
        return channels[0].illustration;
    }
    if (template.description.channel.illustration) {
        return template.description.channel.illustration;
    }

    const boards = template.content.filter((c) => c.board).map((c) => c.board!);
    if (boards.length) {
        return boards[0].illustration;
    }
    if (template.description.board.illustration) {
        return template.description.board.illustration;
    }

    const playbooks = template.content.filter((c) => c.playbook).map((c) => c.playbook!);
    if (playbooks.length) {
        return playbooks[0].illustration;
    }
    if (template.description.playbook.illustration) {
        return template.description.playbook.illustration;
    }

    return '';
}

export const getContentCount = (template: WorkTemplate, playbookTemplates: PlaybookTemplateType[]) => {
    const res = {
        playbooks: 0,
        boards: 0,
    };
    for (const item of template.content) {
        if (item.playbook) {
            const pbTemplate = playbookTemplates.find((pb) => pb.title === item.playbook.template);
            if (pbTemplate) {
                res.playbooks++;
            }
        } else if (item.board) {
            res.boards++;
        }
    }

    return res;
};
