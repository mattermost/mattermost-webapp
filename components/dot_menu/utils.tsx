// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {trackEvent} from 'actions/telemetry_actions';

import {EventTypes, TELEMETRY_CATEGORIES} from 'utils/constants';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (e: ChangeEvent, suffix: string): void => {
    if (e.type === EventTypes.CLICK) {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.CLICK + '_' + suffix);
    } else {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.SHORTCUT + '_ ' + suffix);
    }
};
