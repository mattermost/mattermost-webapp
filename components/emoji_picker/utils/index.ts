// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Emoji, EmojiCategory} from 'mattermost-redux/types/emojis';

import {compareEmojis} from 'utils/emoji_utils';

import {Categories} from '../types';

function sortEmojis(emojis: Emoji[], recentEmojiStrings: string[], filter: string) {
    const recentEmojis: Emoji[] = [];
    const emojisMinusRecent: Emoji[] = [];

    Object.values(emojis).forEach((emoji) => {
        let emojiArray = emojisMinusRecent;
        const alias = 'short_names' in emoji ? emoji.short_names : [emoji.name];
        for (let i = 0; i < alias.length; i++) {
            if (recentEmojiStrings.includes(alias[i].toLowerCase())) {
                emojiArray = recentEmojis;
            }
        }

        emojiArray.push(emoji);
    });

    return [
        ...recentEmojis.sort((firstEmoji, secondEmoji) => compareEmojis(firstEmoji, secondEmoji, filter)),
        ...emojisMinusRecent.sort((firstEmoji, secondEmoji) => compareEmojis(firstEmoji, secondEmoji, filter)),
    ];
}

function getFilteredEmojis(allEmojis: Record<string, Emoji>, filter: string, recentEmojisString: string[]): Emoji[] {
    const emojis = Object.values(allEmojis).filter((emoji) => {
        const alias = 'short_names' in emoji ? emoji.short_names : [emoji.name];
        for (let i = 0; i < alias.length; i++) {
            if (alias[i].toLowerCase().includes(filter)) {
                return true;
            }
        }

        return false;
    });

    return sortEmojis(emojis, recentEmojisString, filter);
}

export function getEmojisByCategory(allEmojis: Record<string, Emoji>, categories: Categories, categoryName: EmojiCategory, filter: string, recentEmojisString: string[]): Emoji[] {
    if (filter.length) {
        return getFilteredEmojis(allEmojis, filter, recentEmojisString);
    }

    // if (categoryName === 'recent') {
    //     const recentEmojiIds = this.state.categories?.recent?.emojiIds ?? [];
    //     if (recentEmojiIds.length === 0) {
    //         return [];
    //     }

    //     const recentEmojis = new Map();

    //     recentEmojiIds.forEach((emojiId) => {
    //         const emoji = this.state.allEmojis[emojiId];
    //         const emojiSkin = getSkin(emoji);

    //         if (emojiSkin) {
    //             const emojiWithUserSkin = this.convertEmojiToUserSkinTone(emoji, emojiSkin, this.props.userSkinTone);
    //             recentEmojis.set(emojiWithUserSkin.unified, emojiWithUserSkin);
    //         } else {
    //             recentEmojis.set(emojiId, emoji);
    //         }
    //     });

    //     return Array.from(recentEmojis.values());
    // }

    return categories[categoryName].emojiIds.map((emojiId) => allEmojis[emojiId]);
}
