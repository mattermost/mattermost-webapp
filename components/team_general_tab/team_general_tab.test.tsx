// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React, {ChangeEvent, ComponentProps} from 'react';
import {shallow} from 'enzyme';

import {TestHelper} from 'utils/test_helper';

import GeneralTab from 'components/team_general_tab/team_general_tab';

describe('components/TeamSettings', () => {
    const getTeam = jest.fn().mockResolvedValue({data: true});
    const patchTeam = jest.fn().mockReturnValue({data: true});
    const regenerateTeamInviteId = jest.fn().mockReturnValue({data: true});
    const removeTeamIcon = jest.fn().mockReturnValue({data: true});
    const setTeamIcon = jest.fn().mockReturnValue({data: true});
    const baseActions = {
        getTeam,
        patchTeam,
        regenerateTeamInviteId,
        removeTeamIcon,
        setTeamIcon,
    };
    const defaultProps: ComponentProps<typeof GeneralTab> = {
        team: TestHelper.getTeamMock({id: 'team_id'}),
        maxFileSize: 50,
        activeSection: 'team_icon',
        updateSection: jest.fn(),
        closeModal: jest.fn(),
        collapseModal: jest.fn(),
        actions: baseActions,
        canInviteTeamMembers: true,
    };

    test('should handle bad updateTeamIcon function call', () => {
        const wrapper = shallow<GeneralTab>(<GeneralTab {...defaultProps}/>);

        wrapper.instance().updateTeamIcon(null as unknown as ChangeEvent<HTMLInputElement>);

        expect(wrapper.state('clientError')).toEqual('An error occurred while selecting the image.');
    });

    test('should handle invalid file selection', () => {
        const wrapper = shallow<GeneralTab>(<GeneralTab {...defaultProps}/>);

        wrapper.instance().updateTeamIcon({
            target: {
                files: [{
                    type: 'text/plain',
                }],
            },
        } as unknown as ChangeEvent<HTMLInputElement>);

        expect(wrapper.state('clientError')).toEqual('Only BMP, JPG or PNG images may be used for team icons');
    });

    test('should handle too large files', () => {
        const wrapper = shallow<GeneralTab>(<GeneralTab {...defaultProps}/>);

        wrapper.instance().updateTeamIcon({
            target: {
                files: [{
                    type: 'image/jpeg',
                    size: defaultProps.maxFileSize + 1,
                }],
            },
        } as unknown as ChangeEvent<HTMLInputElement>);

        expect(wrapper.state('clientError')).toEqual('Unable to upload team icon. File is too large.');
    });

    test('should call actions.setTeamIcon on handleTeamIconSubmit', () => {
        const actions = {...baseActions};
        const props = {...defaultProps, actions};
        const wrapper = shallow<GeneralTab>(<GeneralTab {...props}/>);

        let teamIconFile = null;
        wrapper.setState({teamIconFile, submitActive: true});
        wrapper.instance().handleTeamIconSubmit();
        expect(actions.setTeamIcon).not.toHaveBeenCalled();

        teamIconFile = {file: 'team_icon_file'} as unknown as File;
        wrapper.setState({teamIconFile, submitActive: false});
        wrapper.instance().handleTeamIconSubmit();
        expect(actions.setTeamIcon).not.toHaveBeenCalled();

        wrapper.setState({teamIconFile, submitActive: true});
        wrapper.instance().handleTeamIconSubmit();

        expect(actions.setTeamIcon).toHaveBeenCalledTimes(1);
        expect(actions.setTeamIcon).toHaveBeenCalledWith(props.team?.id, teamIconFile);
    });

    test('should call actions.removeTeamIcon on handleTeamIconRemove', () => {
        const actions = {...baseActions};
        const props = {...defaultProps, actions};
        const wrapper = shallow<GeneralTab>(<GeneralTab {...props}/>);

        wrapper.instance().handleTeamIconRemove();

        expect(actions.removeTeamIcon).toHaveBeenCalledTimes(1);
        expect(actions.removeTeamIcon).toHaveBeenCalledWith(props.team?.id);
    });

    test('hide invite code if no permissions for team inviting', () => {
        const props = {...defaultProps, canInviteTeamMembers: false};

        const wrapper1 = shallow(<GeneralTab {...defaultProps}/>);
        const wrapper2 = shallow(<GeneralTab {...props}/>);

        expect(wrapper1).toMatchSnapshot();
        expect(wrapper2).toMatchSnapshot();
    });

    test('should call actions.patchTeam on handleAllowedDomainsSubmit', () => {
        const actions = {...baseActions};
        const props = {...defaultProps, actions};
        const wrapper = shallow<GeneralTab>(<GeneralTab {...props}/>);

        wrapper.instance().handleAllowedDomainsSubmit();

        expect(actions.patchTeam).toHaveBeenCalledTimes(1);
        expect(actions.patchTeam).toHaveBeenCalledWith(props.team);
    });

    test('should call actions.patchTeam on handleNameSubmit', () => {
        const actions = {...baseActions};
        const props = {...defaultProps, actions};
        if (props.team) {
            props.team.display_name = 'TestTeam';
        }

        const wrapper = shallow<GeneralTab>(<GeneralTab {...props}/>);

        wrapper.instance().handleNameSubmit();

        expect(actions.patchTeam).toHaveBeenCalledTimes(1);
        expect(actions.patchTeam).toHaveBeenCalledWith(props.team);
    });

    test('should call actions.patchTeam on handleInviteIdSubmit', () => {
        const actions = {...baseActions};
        const props = {...defaultProps, actions};
        if (props.team) {
            props.team.invite_id = '12345';
        }

        const wrapper = shallow<GeneralTab>(<GeneralTab {...props}/>);

        wrapper.instance().handleInviteIdSubmit();

        expect(actions.regenerateTeamInviteId).toHaveBeenCalledTimes(1);
        expect(actions.regenerateTeamInviteId).toHaveBeenCalledWith(props.team?.id);
    });

    test('should call actions.patchTeam on handleDescriptionSubmit', () => {
        const actions = {...baseActions};
        const props = {...defaultProps, actions};

        const wrapper = shallow<GeneralTab>(<GeneralTab {...props}/>);

        const newDescription = 'The Test Team';
        wrapper.setState({description: newDescription});
        wrapper.instance().handleDescriptionSubmit();
        if (props.team) {
            props.team.description = newDescription;
        }

        expect(actions.patchTeam).toHaveBeenCalledTimes(1);
        expect(actions.patchTeam).toHaveBeenCalledWith(props.team);
    });

    test('should match snapshot when team is group constrained', () => {
        const props = {...defaultProps};
        if (props.team) {
            props.team.group_constrained = true;
        }

        const wrapper = shallow(<GeneralTab {...props}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should call actions.getTeam on handleUpdateSection if invite_id is empty', () => {
        const actions = {...baseActions};
        const props = {...defaultProps, actions};
        if (props.team) {
            props.team.invite_id = '';
        }

        const wrapper = shallow<GeneralTab>(<GeneralTab {...props}/>);

        wrapper.instance().handleUpdateSection('invite_id');

        expect(actions.getTeam).toHaveBeenCalledTimes(1);
        expect(actions.getTeam).toHaveBeenCalledWith(props.team?.id);
    });
});
