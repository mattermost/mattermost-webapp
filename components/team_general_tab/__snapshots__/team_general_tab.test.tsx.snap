// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TeamSettings hide invite code if no permissions for team inviting 1`] = `
<div>
  <div
    className="modal-header"
  >
    <button
      aria-label="Close"
      className="close"
      data-dismiss="modal"
      id="closeButton"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <h4
      className="modal-title"
    >
      <div
        className="modal-back"
      >
        <span
          onClick={[MockFunction]}
        >
          <BackIcon />
        </span>
      </div>
      <MemoizedFormattedMessage
        defaultMessage="General Settings"
        id="general_tab.title"
      />
    </h4>
  </div>
  <div
    className="user-settings"
  >
    <h3
      className="tab-header"
    >
      <MemoizedFormattedMessage
        defaultMessage="General Settings"
        id="general_tab.title"
      />
    </h3>
    <div
      className="divider-dark first"
    />
    <Connect(SettingItemMin)
      describe="name"
      section="name"
      title="Team Name"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <Connect(SettingItemMin)
      describe=""
      section="description"
      title="Team Description"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <SettingPicture
      clientError=""
      file={null}
      helpText={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Upload a team icon in BMP, JPG or PNG format.\\\\nSquare images with a solid background color are recommended."
          id="setting_picture.help.team"
        />
      }
      imageContext="team"
      loadingPicture={false}
      onFileChange={[Function]}
      onRemove={[Function]}
      onSubmit={[Function]}
      serverError=""
      src={null}
      submitActive={false}
      title="Team Icon"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <Connect(SettingItemMin)
      describe=""
      section="allowed_domains"
      title="allowedDomains"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <OpenInvite
      allowOpenInvite={false}
      isActive={false}
      isGroupConstrained={false}
      onToggle={[Function]}
      patchTeam={[MockFunction]}
      teamId="team_id"
    />
    <div
      className="divider-light"
    />
    <Connect(SettingItemMin)
      describe="Click 'Edit' to regenerate Invite Code."
      section="invite_id"
      title="Invite Code"
      updateSection={[Function]}
    />
    <div
      className="divider-dark"
    />
  </div>
</div>
`;

exports[`components/TeamSettings hide invite code if no permissions for team inviting 2`] = `
<div>
  <div
    className="modal-header"
  >
    <button
      aria-label="Close"
      className="close"
      data-dismiss="modal"
      id="closeButton"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <h4
      className="modal-title"
    >
      <div
        className="modal-back"
      >
        <span
          onClick={[MockFunction]}
        >
          <BackIcon />
        </span>
      </div>
      <MemoizedFormattedMessage
        defaultMessage="General Settings"
        id="general_tab.title"
      />
    </h4>
  </div>
  <div
    className="user-settings"
  >
    <h3
      className="tab-header"
    >
      <MemoizedFormattedMessage
        defaultMessage="General Settings"
        id="general_tab.title"
      />
    </h3>
    <div
      className="divider-dark first"
    />
    <Connect(SettingItemMin)
      describe="name"
      section="name"
      title="Team Name"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <Connect(SettingItemMin)
      describe=""
      section="description"
      title="Team Description"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <SettingPicture
      clientError=""
      file={null}
      helpText={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Upload a team icon in BMP, JPG or PNG format.\\\\nSquare images with a solid background color are recommended."
          id="setting_picture.help.team"
        />
      }
      imageContext="team"
      loadingPicture={false}
      onFileChange={[Function]}
      onRemove={[Function]}
      onSubmit={[Function]}
      serverError=""
      src={null}
      submitActive={false}
      title="Team Icon"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <Connect(SettingItemMin)
      describe=""
      section="allowed_domains"
      title="allowedDomains"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <OpenInvite
      allowOpenInvite={false}
      isActive={false}
      isGroupConstrained={false}
      onToggle={[Function]}
      patchTeam={[MockFunction]}
      teamId="team_id"
    />
    <div
      className="divider-light"
    />
    <div
      className="divider-dark"
    />
  </div>
</div>
`;

exports[`components/TeamSettings should match snapshot when team is group constrained 1`] = `
<div>
  <div
    className="modal-header"
  >
    <button
      aria-label="Close"
      className="close"
      data-dismiss="modal"
      id="closeButton"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <h4
      className="modal-title"
    >
      <div
        className="modal-back"
      >
        <span
          onClick={[MockFunction]}
        >
          <BackIcon />
        </span>
      </div>
      <MemoizedFormattedMessage
        defaultMessage="General Settings"
        id="general_tab.title"
      />
    </h4>
  </div>
  <div
    className="user-settings"
  >
    <h3
      className="tab-header"
    >
      <MemoizedFormattedMessage
        defaultMessage="General Settings"
        id="general_tab.title"
      />
    </h3>
    <div
      className="divider-dark first"
    />
    <Connect(SettingItemMin)
      describe="TestTeam"
      section="name"
      title="Team Name"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <Connect(SettingItemMin)
      describe="The Test Team"
      section="description"
      title="Team Description"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <SettingPicture
      clientError=""
      file={null}
      helpText={
        <Memo(MemoizedFormattedMessage)
          defaultMessage="Upload a team icon in BMP, JPG or PNG format.\\\\nSquare images with a solid background color are recommended."
          id="setting_picture.help.team"
        />
      }
      imageContext="team"
      loadingPicture={false}
      onFileChange={[Function]}
      onRemove={[Function]}
      onSubmit={[Function]}
      serverError=""
      src={null}
      submitActive={false}
      title="Team Icon"
      updateSection={[Function]}
    />
    <div
      className="divider-light"
    />
    <OpenInvite
      allowOpenInvite={false}
      isActive={false}
      isGroupConstrained={true}
      onToggle={[Function]}
      patchTeam={[MockFunction]}
      teamId="team_id"
    />
    <div
      className="divider-dark"
    />
  </div>
</div>
`;
